File: 1620-check-if-array-pairs-are-divisible-by-k/1620-check-if-array-pairs-are-divisible-by-k.py
class Solution:
    def canArrange(self, arr: List[int], k: int) -> bool:
        result=[0]*k

        for num in arr:
            result[num%k]+=1
        for r in range(k):
            if r==0:
                if result[r]%2!=0:
                    return False

            else:
                if result[r]!=result[k-r]:
                    return False
        return True
            
    
        

File: 1620-check-if-array-pairs-are-divisible-by-k/1620-check-if-array-pairs-are-divisible-by-k.java
class Solution {
    public boolean canArrange(int[] arr, int k) {
        
        int [] count=new int [k];
        
        for (int num: arr){
            int remainder = (num % k + k) % k;
            count[remainder]++;
        }

        for( int i=0;i<k;i++){
            if(i==0){
                if(count[i]%2!=0){
                    return false;
                }
            }
            else{
                if(k-i>=0){

                if(count[i]!=count[k-i]){
                    return false;
                }
                }
            }
        }
        return true;
    }
}